lang/src/org/partiql/lang/syntax/SqlParser.kt

```diff
@@ -708,9 +708,9 @@ class SqlParser(private val ion: IonSystem) : Parser {
             }
             TIME -> {
                 val timeString = token!!.text!!
-                val precision = children[0].token?.value?.numberValue()?.toInt()
+                val precision = children[0].token!!.value!!.numberValue().toInt()
                 val time = LocalTime.parse(timeString, DateTimeFormatter.ISO_TIME)
```

Comment:
I wasn't being clear--I think the `?` is fine, but you probably want a `?: throw something` at the end of this to get a better exception unless you really want the NPE.